src/tests/ftest/checksum/basic_checksum.py

```diff
@@ -0,0 +1,153 @@
+        self.pool = None
+        self.container = None
+        self.obj = None
+        self.ioreq = None
+        self.no_of_dkeys = None
+        self.no_of_akeys = None
+        self.array_size = None
+        self.record_length = None
+
+        self.no_of_dkeys = self.params.get("no_of_dkeys", '/run/dkeys/*')[0]
+        self.no_of_akeys = self.params.get("no_of_akeys", '/run/akeys/*')[0]
+        self.record_length = self.params.get("length", '/run/record/*')
+
+        self.pool = TestPool(self.context)
+        self.pool.get_params(self)
+        self.pool.create()
+        self.pool.connect(2)
+        self.csum = self.params.get("enable_checksum", '/run/container/*')
+        self.container = DaosContainer(self.context)
+        self.container.create(poh=self.pool.pool.handle,
+                              enable_chksum=self.csum)
+        self.container.open()
+
+        self.obj = DaosObj(self.context, self.container)
+        self.obj.create(objcls=1)
+        self.obj.open()
+        self.ioreq = IORequest(self.context,
+                               self.container,
+                               self.obj, objtype=4)
+
+    def reconnect(self):
+        '''
+        Function to reconnect the pool/container and reopen the Object
+        for read verification.
+        '''
+        # Close the Obj/Container, Disconnect the Pool.
+        self.obj.close()
+        self.container.close()
+        self.pool.disconnect()
+        time.sleep(5)
+        # Connect Pool, Open Container and Object
+        self.pool.connect(2)
+        self.container.open()
+        self.obj.open()
+        self.ioreq = IORequest(self.context,
+                               self.container,
+                               self.obj,
+                               objtype=4)
+
+    def test_single_object_with_checksum(self):
+        """
+        Test ID: DAOS-3927
+        Test Description: Write Avocado Test to verify single data after
+                          pool/container disconnect/reconnect.
+        :avocado: tags=all,full_regression,pr,basic_checksum_object
+        """
+        self.d_log.info("Writing the Single Dataset")
+        record_index = 0
+        transaction = []
+        for dkey in range(self.no_of_dkeys):
+            for akey in range(self.no_of_akeys):
+                indata = ("{0}".format(str(akey)[0])
+                          * self.record_length[record_index])
+                c_dkey = ctypes.create_string_buffer("dkey {0}".format(dkey))
+                c_akey = ctypes.create_string_buffer("akey {0}".format(akey))
+                c_value = ctypes.create_string_buffer(indata)
+                c_size = ctypes.c_size_t(ctypes.sizeof(c_value))
+
+                new_transaction = self.container.get_new_tx()
+                self.ioreq.single_insert(c_dkey, c_akey, c_value, c_size,
+                                         new_transaction)
+                self.container.commit_tx(new_transaction)
```

Comment:
We should really change the python API not to make the transaction mandatory. single_insert (and other insert function) should take a transaction=none param and not require a get_new_tx() commit_tx() call for those simple cases.

Reply:
Some of the implementation are in existing tests which I used for this basic test.

Reply:
ok, but please do it shortly after landing this PR.